File: pset6/vigenere/vigenere.py
#!/usr/bin/env python3

import sys
from cs50 import get_string

# validation of command line args


def comandLineArgsValidation():
    # check for number of arguments
    if len(sys.argv) != 2:
        print("Usage ./vigenere k")
        return False

    # check for arg contains only aplphabet symbols
    if sys.argv[1].isalpha() == False:
        return False
    return True

# text encryption


def encryptText(plainText, key):

    # convert params strins to lists
    plainTextAsList = list(plainText)
    keyAsList = list(key)
    keyIndex = 0
    k = 0
    # text encryption
    for i in range(len(plainTextAsList)):
        # calculating of actual key value
        if keyAsList[keyIndex].isupper() == True:
            k = ord(keyAsList[keyIndex]) - ord('A')
        else:
            k = ord(keyAsList[keyIndex]) - ord('a')
        # for upper case
        if plainTextAsList[i].isalpha():
            if plainTextAsList[i].isupper():
                plainTextAsList[i] = chr((((ord(plainTextAsList[i]) - ord('A')) + k) % 26) + ord('A'))
            # for lower case
            elif plainTextAsList[i].islower():
                plainTextAsList[i] = chr((((ord(plainTextAsList[i]) - ord('a')) + k) % 26) + ord('a'))
            keyIndex += 1
            if keyIndex == len(keyAsList):
                keyIndex = 0
    plainText = ""
    plainText = ''.join(plainTextAsList)

    return plainText


# check command line args
if comandLineArgsValidation() == False:
    exit(1)

# get plain text
plainText = get_string("plaintext: ")

# call encryption function
plainText = encryptText(plainText, sys.argv[1])

print("ciphertext: {}".format(plainText))

File: pset2/vigenere/vigenere.c
// basic vigenere chipher encryption realisation
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

void encryptText(char *plainText, char *key);

bool comandLineArgsValidation(int argc, char **argv);

int main(int argc, char **argv)
{
    // check command line args
    if (!comandLineArgsValidation(argc, argv))
    {
        return 1;
    }

    // get plain text
    string plainText = get_string("plaintext: ");

    // call encryption function
    encryptText(plainText, argv[1]);

    printf("ciphertext: %s\n", plainText);

    return 0;
}

// text encryption
void encryptText(char *plainText, char *key)
{
    // init key params
    int keyIndex = 0;
    int keyLength = strlen(key);

    // actual key
    int k;

    for (int i = 0, l = strlen(plainText); i < l; i++)
    {
        // calculating of actual key value
        if (isupper(key[keyIndex]))
        {
            k = key[keyIndex] - 'A';
        }
        else
        {
            k = key[keyIndex] - 'a';
        }

        // onlu for alphabet characters
        if (isalpha(plainText[i]))
        {
            // for upper case
            if (isupper(plainText[i]))
            {
                plainText[i] = (((plainText[i] - 'A') + k) % 26) + 'A';
            }
            // for lower case
            else if (islower(plainText[i]))
            {
                plainText[i] = (((plainText[i] - 'a') + k) % 26) + 'a';
            }

            // iteration of keys
            if (++keyIndex == keyLength)
            {
                keyIndex = 0;
            }
        }
    }
}

// validation of command line args
bool comandLineArgsValidation(int argc, char **argv)
{
    // check for number of arguments
    if (argc != 2)
    {
        printf("Usage ./vigenere k\n");
        return false;
    }

    // check for arg  conatins only alphabet symblos
    for (int i = 0, l = strlen(argv[1]); i < l; i++)
    {
        if (!isalpha(argv[1][i]))
        {
            printf("Usage ./vigenere k\n");
            return false;
        }
    }
    return true;
}